src/agent.cpp

```diff
@@ -249,62 +277,68 @@ void Agent::actionImpl() {
   }
 
   // For now let's not worry about turning the neck or setting the vision.
-  // However, do this now so doesn't override anything changed by the requested action.
-  // TODO for librcsc: setViewActionDefault, setNeckActionDefault that will not overwrite if already set.
+  // But do the settings now, so that doesn't override any set by the actions below.
+  // TODO: Add setViewActionDefault, setNeckActionDefault to librcsc that only set if not already set.
   this->setViewAction(new View_Tactical());
   this->setNeckAction(new Neck_TurnToBallOrScan());
 
+  const WorldModel & wm = this->world();
+  bool may_fix = wm.self().collidesWithPost();
```

Comment:
Heh. I would guess there's at least one linter out there for C++ that could give McCabe (cyclomatic) complexity; pylint can for python with the right plugin. While working on adding some utilities to hfo.py (in my fork's state_parsing_functions branch), I've tried to make pylint a bit happier with that code. (BTW, should utility functions be in hfo.py, or perhaps in a separate util.py module?)

OK. Should an UNKNOWN return from the fetching-feedback function still be possible, if the program "asks" for an action that isn't the last one? Or give an error message? I do think that it is a good idea to doublecheck what action the feedback is meant for, given potential message losses and confusion given timing re act, step, etc...